opencti-platform/opencti-graphql/tests/02-integration/01-database/stix-filtering-testers-test.ts

```diff
@@ -0,0 +1,670 @@
+import { describe, expect, it } from 'vitest';
+
+// basic set
+// import stixBundle from '../../data/DATA-TEST-STIX2_v2.json';
+
+// specific data object that are not covered by the basic set, all recorded from real stream events
+import stixReports from '../../data/stream-events/stream-event-stix2-reports.json';
+import stixIndicators from '../../data/stream-events/stream-event-stix2-indicators.json';
+import stixIncidents from '../../data/stream-events/stream-event-stix2-incidents.json';
+import stixRfis from '../../data/stream-events/stream-event-stix2-rfis.json';
+import stixSightings from '../../data/stream-events/stream-event-stix2-sightings.json';
+import stixRelationships from '../../data/stream-events/stream-event-stix2-relationships.json';
+
+import * as testers from '../../../src/utils/stix-filtering/stix-testers';
+import type { Filter } from '../../../src/utils/stix-filtering/filter-group';
+
+describe('Stix filter testers', () => {
+  describe('by Markings (key=markedBy)', () => {
+    const stixWithMarkings = stixIndicators[0];
+    const stixWithoutMarkings = stixReports[0];
+
+    it('should test positive for a stix object with matching filter', () => {
+      let filter: Filter = {
+        key: ['markedBy'],
+        mode: 'OR',
+        operator: 'eq',
+        values: ['<some-id>', 'marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9']
+      };
+      expect(testers.testMarkingFilter(stixWithMarkings, filter)).toEqual(true);
+      expect(testers.testMarkingFilter(stixWithoutMarkings, filter)).toEqual(false);
+
+      filter = {
+        key: ['markedBy'],
+        mode: 'AND',
+        operator: 'eq',
+        values: ['<some-id>']
+      };
+      expect(testers.testMarkingFilter(stixWithMarkings, filter)).toEqual(false);
+      expect(testers.testMarkingFilter(stixWithoutMarkings, filter)).toEqual(false);
+    });
+  });
+
+  describe('by Entity Type (key=entity_type)', () => {
+    const report = stixReports[0];
+    const stixWithExtType = stixIndicators[0];
+
+    it('should test positive for a stix object with matching filter, using generated internal type', () => {
+      let filter: Filter = {
+        key: ['entity_type'],
```

Comment:
Remark: for me (feed, dynamic filters etc), backend tests are also working if the key is a string (instead of an array of strings)

Reply:
The type for keys on your side is `string | string[]` ?

Reply:
The type is an array 
![image](https://github.com/OpenCTI-Platform/opencti/assets/75783086/8747e4cb-569a-4cb5-838b-0b6aee557485)

Reply:
but string are also handled, thus my tests are working...

Reply:
(old behavior)

Reply:
You can keep your behavior, it was just a remark, for now, the back handle both (from old behavior)